### PR TITLE
Refactor compilation workflow to support CPU without CUDA

### PR DESCRIPTION
**Pull request by LDY1998**

### CODE REVIEW

**Comment by yaoyaoding** [comment]
Thanks @LDY1998! I also updated part of the implementation, could you have a look?

**Review by LDY1998 — COMMENTED** [review]
Thanks @yaoyaoding for the update. I only left a very minor comment. I think overall it's good to go.

File: python/hidet/backend/build.py [hunk 1]
```diff
@@ -39,20 +39,163 @@ def __str__(self):
...
+                # if the compilation failed, raise an exception
+                if result.returncode:
+                    message = "Command: {}\n".format(command)
+                    if result.stdout:
+                        message += result.stdout.decode().strip() + '\n'
+                    if result.stderr:
+                        message += result.stderr.decode().strip()
+                    raise CompilationFailed(src_path, message)
+
+                # write the compilation log to "nvcc_log.txt"
```
Comment (LDY1998) [inline python/hidet/backend/build.py:74]: Do we want to leave the comment like that

**Comment by yaoyaoding** [comment]
Thanks @LDY1998.

**Pull request merged**